src/Lib.P2P/Multiplex/Muxer.cs

```diff
@@ -0,0 +1,340 @@
+                Id = streamId,
+                Name = name,
+                Muxer = this,
+                SentMessageType = PacketType.MessageInitiator,
+            };
+            Substreams.TryAdd(streamId, substream);
+
+            // Tell the other side about the new stream.
+            using (await AcquireWriteAccessAsync().ConfigureAwait(false))
+            {
+                var header = new Header {StreamId = streamId, PacketType = PacketType.NewStream};
+                var wireName = Encoding.UTF8.GetBytes(name);
+                await header.WriteAsync(Channel, cancel).ConfigureAwait(false);
+                await Channel.WriteVarintAsync(wireName.Length, cancel).ConfigureAwait(false);
+                await Channel.WriteAsync(wireName, 0, wireName.Length, cancel).ConfigureAwait(false);
+                await Channel.FlushAsync(cancel).ConfigureAwait(false);
+            }
+
+            return substream;
+        }
+
+        /// <summary>
+        ///   Remove the stream.
+        /// </summary>
+        /// <remarks>
+        ///   Internal method called by Substream.Dispose().
+        /// </remarks>
+        public async Task<Substream> RemoveStreamAsync(Substream stream,
+            CancellationToken cancel = default)
+        {
+            if (Substreams.TryRemove(stream.Id, out _))
+
+                // Tell the other side.
+                using (await AcquireWriteAccessAsync().ConfigureAwait(false))
+                {
+                    var header = new Header
+                    {
+                        StreamId = stream.Id,
+                        PacketType = PacketType.CloseInitiator
+                    };
+                    await header.WriteAsync(Channel, cancel).ConfigureAwait(false);
+                    Channel.WriteByte(0); // length
+                    await Channel.FlushAsync(cancel).ConfigureAwait(false);
+                }
+
+            return stream;
+        }
+
+        /// <summary>
+        ///   Read the multiplex packets.
+        /// </summary>
+        /// <param name="cancel"></param>
+        /// <returns></returns>
+        /// <remarks>
+        ///   A background task that reads and processes the multiplex packets while
+        ///   the <see cref="Channel"/> is open and not <paramref name="cancel">cancelled</paramref>.
+        ///   <para>
+        ///   Any encountered errors will close the <see cref="Channel"/>.
+        ///   </para>
+        /// </remarks>
+        public async Task ProcessRequestsAsync(CancellationToken cancel = default)
+        {
+            try
+            {
+                while (Channel.CanRead && !cancel.IsCancellationRequested)
+                {
+                    // Read the packet prefix.
+                    var header = await Header.ReadAsync(Channel, cancel).ConfigureAwait(false);
+                    var length = await Varint.ReadVarint32Async(Channel, cancel).ConfigureAwait(false);
+                    if (_log.IsTraceEnabled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)